test/inductor/test_triton_kernels.py

```diff
@@ -3473,11 +3433,72 @@ def add(x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
         assert add(x, y).mean() == 2, "Problem with add kernel"
 
         # this should cause an exception, since pre_hook is not allowed
-        msg = "pre_hook is not supported in triton.Autotune Configs"
+        msg = "pre_hook and post_hook are not supported in triton.Autotune or triton.Config"
         with self.assertRaisesRegex(torch._dynamo.exc.Unsupported, msg):
             add_compiled = torch.compile(add, mode="reduce-overhead", fullgraph=True)
             add_compiled(x, y).mean()
 
+    # we have to enable autotune at compile time for this test
+    @requires_gpu
+    @common_utils.parametrize("backend", ["eager", "aot_eager", "inductor"])
+    @common_utils.parametrize("autotune_at_compile_time", [True])
+    def test_triton_kernel_reset_to_zero(self, backend, autotune_at_compile_time):
+        if autotune_at_compile_time and backend != "inductor":
+            raise unittest.SkipTest("compile-time autotuning only exists in inductor")
+
+        @triton.autotune(
+            configs=[
+                triton.Config(
+                    {"BLOCK_SIZE": 64, "COND": 1234}, num_stages=3, num_warps=8
+                ),
+                triton.Config(
+                    {"BLOCK_SIZE": 32, "COND": 1234}, num_stages=3, num_warps=8
+                ),
+                triton.Config(
+                    {"BLOCK_SIZE": 16, "COND": 1234}, num_stages=3, num_warps=8
+                ),
+            ],
+            key=[],
+            reset_to_zero=["counter"],
+        )
+        @triton.jit
+        def increment_kernel(
+            in_ptr0,
+            counter,  # reset this to zero every time
```

Comment:
Nit: the name `counter` for this param sounds confusing. Could we rename it to `increment_ptr` or something? Also for the related vars below.